user-service/domains/repositories/user_repository.py

```diff
@@ -96,5 +96,36 @@ def add_following(self, user_id, following=[]):
         self.session.merge(user)
         self.session.commit()
         return user
+    """
+    Helper for updating a User object
 
+    :param username
+    :return: 
+    """
+    @check_id_exists(User, ["user_id", "username"])
+    @check_unique(User, User.username, ["username"])
+    def _update_user(self, user: User, username, streaming_status):
```

Comment:
just pass in user and commit the changes, don't modify username and streaming_status for the user here, do it in update_user

Reply:
for this, i needed to do a helper function to double check username because for the main function, i can't make sure username exists and is unique, since you wanted me to not do anything if username isn't passed, not throw an error

Reply:
I see, don't use the decorator then. Just check whether username if a username object is passed and if so, check whether it exists programmatically and throw and error if it does.

Reply:
okay, modified, please check again